### DISCUSSION TITLE
is it possible to migrate 1.4.2 to 3.0?

### DISCUSSION THREAD

**Discussion by ghost** [uswds/uswds#4895] [General]
Good afternoon!

I am new with uswds theme. Our website uses uswds as our base theme, recently we noticed that we cannot implement some components to our site by following the instruction online. I just checked our site uses uswds 1.4.2, probably it causes the issues. 

I need to migrate it to the latest version 3.0. Is it possible for me to jump from 1.4.2 to 3.0? What all necessary changes do I need in our codes?

Thanks so much for your time and help!

**Comment by mejiaj** [comment]
@xiaohuima-2020 we've created an [issue (#1733)](https://github.com/uswds/uswds-site/issues/1733) based on your [comment](https://github.com/uswds/uswds/discussions/4660#discussioncomment-3300774) and this discussion. We will add a note for those on USWDS 1x.

I'd recommend
1. Identify if you're using a Drupal USWDS base theme (if so, update)
1. Look over [migration notes](https://designsystem.digital.gov/documentation/migration-v2/#summary) from 1x → 2x
   1. Update SASS starting with [Mixins and functions](https://designsystem.digital.gov/documentation/migration-v2/#mixins-and-functions)
   1. Update [component classes](https://designsystem.digital.gov/documentation/migration-v2/#migration-by-component )
1. Follow [migration to 3](https://designsystem.digital.gov/documentation/migration/) this should be more straightforward than 1x → 2x

We'd love for you to share your experience so we can improve the docs.

**Reply by ghost** [reply]
Thank you for your quickly response!

Our Drupal website uses USWDS base theme 1.4.2. 
From my understanding on No.2, I need make all necessary changes (listed in migration notes from 1->2) in scss files for our sub theme. Is it right?

Thanks again!

**Reply by mejiaj** [reply]
@xiaohuima-2020 yes, after updating your base theme.

**Reply by ghost** [reply]
Thank you!

After checking our base theme directories, I found our web site uses theme from https://www.drupal.org/project/uswds instead of getting it from github. Please let me know if I should get the latest version, https://www.drupal.org/project/uswds/releases/8.x-2.5. 

Thanks again!

**Reply by mejiaj** [reply]
@xiaohuima-2020 that base Drupal theme has classes and templates to allow you to integrate USWDS more easily.

You **still** need the official USWDS package.
